### PR TITLE
net-snmp: add perl variants

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.4 21F79 x86_64
Xcode 13.4 13F17a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: net/net-snmp/Portfile [hunk 1]
```diff
@@ -27,7 +27,9 @@ long_description        This is net-snmp, a derivative of CMU's SNMP \
                         handle SNMP traps, a version of the unix 'netstat' \
                         command using SNMP, a Tk/perl MIB browser.
 
-perl5.branches          5.28
+perl5.branches          5.28 5.30 5.32 5.34
+perl5.default_branch    5.32
```
Comment (ryandesign) [inline net/net-snmp/Portfile:31]: Why default to 5.32? 5.28 is the preferred default per https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/perl5-1.0.tcl#L17.

**Review by ryandesign — COMMENTED** [review]
Why are perl variants desired?

**Comment by i0ntempest** [comment]
> Why are perl variants desired?

To reduce clutters. My system has 4 Perl versions installed due to ports using different versions of them. Also, mismatching Perl versions cause build failure with po4a (say you have perl5 +5_34 installed and po4a by default uses 5.30 iirc).
As for the default version, I’ll put them back to 5.28.

**Comment by ryandesign** [comment]
Reducing the number of installed perl versions is exactly why we specify a preferred default perl version. The only ports that use a perl other than 5.28 should be those that are incompatible with 5.28. Of course we should use a newer perl by default in the near future but that hasn't been done yet.

If po4a experiences a build failure when perl5 is installed with a different variant then that is a bug in po4a; file a ticket.

**Comment by i0ntempest** [comment]
Just put the default back to 5.28. Users should still have the choice though, I tend to make everything to use the newest Perl.

**Pull request merged**